index.js

```diff
@@ -757,6 +758,20 @@ class KeyringController extends EventEmitter {
     this.memStore.updateState({ isUnlocked: true });
     this.emit('unlock');
   }
+
+  /**
+   * Forget hardware keyring
+   *
+   * Forget hardware and update memorized state.
+   * @param {Keyring} keyring
+   */
+  forgetKeyring(keyring) {
+    if (keyring.forgetDevice) {
+      keyring.forgetDevice();
+      this.persistAllKeyrings.bind(this)();
+      this._updateMemStoreKeyrings.bind(this)();
+    }
```

Comment:
Could you throw an error here, if `forgetDevice` doesn't exist? Otherwise it might appear that it worked.

Reply:
Cool

Reply:
@Gudahtt The code was updated by adding an else.